### PR TITLE
Updated to PHP CodeSniffer 3

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Awesome job! I'm amazed you think about future compatibility in such level.

**Comment by TomasVotruba** [comment]
Let me know if you need any help.

**Comment by TomasVotruba** [comment]
Showcase PR with CodeSniffer upgrade 2.x => 3.x: https://github.com/hostnet/phpcs-tool/pull/16/files

**Comment by TomasVotruba** [comment]
Hey, I just wanted to ask. I depend with few packages of mine on this, using awesome helpers in it + CodeSniffer 3.0. Among 2 from Symplify, also https://github.com/object-calisthenics/phpcs-calisthenics-rules

This is the last unupdated-package to get to stable release, that I'd like to sync with Symfony 3.3 release (in ~10 days). **I'd like to know, when do you plan to merge and release CodeSniffer 3.0 version.**

So I could decide, if I should wait or if I should copy-paste those helpers for now and update them to CodeSniffer 3.0.

**Comment by VasekPurchart** [comment]
CodeSniffer in versions `2.9.1` and `3.0` is unusable for code on PHP 7.1 using nullable types, see https://github.com/squizlabs/PHP_CodeSniffer/pull/1442, so there is not much sense in supporting it currently unfortunately.

**Comment by TomasVotruba** [comment]
@VasekPurchart Thanks for linking this. I hope this will be solved ASAP.

File: composer.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
 	},
 	"require": {
 		"php": "^7.0",
-		"squizlabs/php_codesniffer": "~2.8.1"
+		"squizlabs/php_codesniffer": "dev-master"
```
Comment (il-m-yamagishi) [inline composer.json]: Hey, I also need this PR to be merged.

But my custom sniff repository currently depends on `"squizlabs/php_codesniffer": "~3.0.0"` .

So I cannot require this branch(also future merged branch).

I suggest the dependency version `~3.0.0` or `^3.0` will be better than `dev-master` .

(I say Sorry when you supposed to change this version when PHPCS3.0.1 released.)

Thanks.

Comment (kukulich) [inline composer.json]: Yes, I'll use ^3.0.1 :) I just want to test that there is no other bug in PHPCS.

**Comment by mavimo** [comment]
[issues](https://github.com/squizlabs/PHP_CodeSniffer/pull/1442) is fixed with phpcs 3.0.1, I think whould possibile use this patch version as minimum requirement.

**Comment by TomasVotruba** [comment]
@kukulich :clap:  :clap:  :clap:

**Pull request merged**